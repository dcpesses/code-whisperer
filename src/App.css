/* variables */
:root{
  --text-primary: #fff;
  --bg-primary: #212529;
  --bg-secondary: #13111c;

  /* component override examples */
  --todo-bg-form: #96c!important;
  --todo-bg-item: #69c!important;
}

/* reset */
* {
  box-sizing: border-box;
}
body, p, ul, li {
  margin: 0;
  padding: 0;
}

/* base styles */
body {
  background-color: var(--bg-secondary);
  color: var(--text-primary);
  min-height: 100vh;
}
.h1, .h2, .h3, .h4, .h5, .h6,
h2, h3, h4, h5, h6 {
  margin-top: 0;
  margin-bottom: .5rem;
  font-weight: 500;
  line-height: 1.2;
}
button {
  cursor: pointer;
}

.App {
  text-align: center;
}

.App-header {
  background-color: var(--bg-primary);
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(8px + 0.8vmin);
  color: var(--text-primary);
}
.App-header p {
  margin-bottom: 1rem;
}

.App-subheader {
  font-size: calc(10px + 1.25vmin);
  border-bottom: 1px solid #495057;
  color: #dee2e6;
  font-size: 16px;
  text-align: left;
  padding-bottom: 0.5rem;
}

.App-logo {
  height: 5vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-logo + .h2 {
  display: inline-block;
  font-size: calc(10px + 1.5vmin);
  line-height: 5vmin;
  vertical-align: top;
  padding-right: 2vmin;
  margin: 0;
  font-weight: bold;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
.App-examples {
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-left: auto;
  margin-right: auto;
}

.App-examples > div {
  display: grid;
  align-items: flex-start;
  justify-content: space-evenly;
  grid-template-columns: repeat(1, minmax(0, 1fr));
  gap: 1rem;
  margin-left: 1rem;
  margin-right: 1rem;
}

.App-example {
  display:flex;
  flex-direction: column;
  min-width: 100%;
  width: 80vw;
  box-sizing: border-box;
  box-shadow: 0px 0px 16px rgba(40,97,230, 0.2);
  background-color: rgb(33, 37, 41, 0.8);
  border: 1px solid rgb(40, 97, 230, 0.8);
  padding: calc(0.5rem + 0.5vmin);
  border-radius: 0.5rem;
  padding: 1rem;
}

@media (min-width: 960px) {
  .App-examples > div {
    grid-template-columns: repeat(3, minmax(0, 1fr));
    grid-auto-rows: 1fr;
  }
  .App-example {
    height: 100%;
    width: 100%;
  }
}
