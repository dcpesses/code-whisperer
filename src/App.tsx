import { Routes, Route } from 'react-router-dom';

// Pages
import AuthenticatedApp from '@/pages/authenticated-app';
import Error404 from '@/pages/error404';
import Login from '@/features/login';
import Landing from '@/pages/landing';

import 'bootstrap/dist/css/bootstrap.css';
import 'bootstrap-icons/font/bootstrap-icons.css';
import '@/App.css';

declare global {
  interface Window {
    lastUpdated: string;
  }
}

function App() {
  return (
    <Routes>
      <Route path="/login" element={<Login />} />
      <Route path="/landing" element={<Landing />} />
      <Route path="/error" element={<Error404 />} />
      <Route path="/*" element={<AuthenticatedApp />} />
    </Routes>
  );
}

export default App;
export {Error404};
