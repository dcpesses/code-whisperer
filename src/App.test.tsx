import { fireEvent, render, screen } from '@testing-library/react';

import App from '@/App';

test('count is 0 at first', () => {
  render(<App />);

  expect(screen.getByText('Count is: 0')).toBeDefined();
});

test('increments count', () => {
  render(<App />);

  fireEvent.click(screen.getByRole('button'));

  expect(screen.getByText('Count is: 1')).toBeDefined();
});
