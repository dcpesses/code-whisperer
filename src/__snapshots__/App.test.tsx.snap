// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Should render as expected 1`] = `
<div>
  <div
    class="App"
  >
    <header
      class="App-header"
    >
      <p>
        <img
          alt="logo"
          class="App-logo"
          src="/vite-react-ts-gh/src/assets/logo.svg"
        />
        <span
          class="h2"
        >
          Hello Vite + React!
        </span>
      </p>
      <p>
        Edit 
        <code>
          App.tsx
        </code>
         and save to test HMR updates.
      </p>
      <p>
        <a
          class="App-link"
          href="https://reactjs.org"
          rel="noopener noreferrer"
          target="_blank"
        >
          Learn React
        </a>
         | 
        <a
          class="App-link"
          href="https://vitejs.dev/guide/features.html"
          rel="noopener noreferrer"
          target="_blank"
        >
          Vite Docs
        </a>
      </p>
    </header>
    <h1>
      Examples
    </h1>
    <div
      class="App-examples"
    >
      <div>
        <div
          class="App-example"
        >
          <h3
            class="App-subheader"
          >
            Redux To-Dos
          </h3>
          <div
            class="todos"
          >
            <div
              class="add-todo"
            >
              <input
                placeholder="Enter a Todo..."
                type="text"
                value=""
              />
              <button>
                Add Todo
              </button>
            </div>
            <ul
              class="todos-list"
            >
              <li
                class="todo-list-item"
              >
                <span
                  class="item-checkbox"
                >
                  <input
                    title="Mark Complete"
                    type="checkbox"
                  />
                </span>
                <span
                  class="item-content"
                  style="text-decoration: none;"
                >
                  Hit the gym
                </span>
                <span
                  class="item-actions"
                >
                  <button
                    class="item-actions-edit"
                    title="Edit"
                  >
                    ✎
                  </button>
                  <button
                    class="item-actions-delete"
                    title="Delete"
                  >
                    ✖
                  </button>
                </span>
              </li>
              <li
                class="todo-list-item"
              >
                <span
                  class="item-checkbox"
                >
                  <input
                    title="Mark Incomplete"
                    type="checkbox"
                  />
                </span>
                <span
                  class="item-content"
                  style="text-decoration: line-through;"
                >
                  Meet George
                </span>
                <span
                  class="item-actions"
                >
                  <button
                    class="item-actions-edit"
                    title="Edit"
                  >
                    ✎
                  </button>
                  <button
                    class="item-actions-delete"
                    title="Delete"
                  >
                    ✖
                  </button>
                </span>
              </li>
            </ul>
          </div>
        </div>
        <div
          class="App-example"
        >
          <h3
            class="App-subheader"
          >
            Redux Counter
          </h3>
          <div>
            <div
              class="_row_503e3a"
            >
              <button
                aria-label="Decrement value"
                class="_button_503e3a"
              >
                 - 
              </button>
              <span
                aria-label="Counter value"
                class="_value_503e3a"
              >
                0
              </span>
              <button
                aria-label="Increment value"
                class="_button_503e3a"
              >
                 + 
              </button>
            </div>
            <div
              class="_row_503e3a"
            >
              <input
                aria-label="Set increment amount"
                class="_textbox_503e3a"
                value="2"
              />
              <button
                aria-label="Add increment amount"
                class="_button_503e3a"
              >
                Add
              </button>
              <div>
                <button
                  aria-label="Add async increment amount"
                  class="_asyncButton_503e3a"
                >
                  Add Async
                </button>
                <button
                  aria-label="Add increment amount if odd"
                  class="_button_503e3a"
                >
                  Add If Odd
                </button>
              </div>
            </div>
          </div>
        </div>
        <div
          class="App-example"
        >
          <h3
            class="App-subheader"
          >
            State Counter
          </h3>
          <div>
            <div
              class="_row_e43be3"
            >
              <button
                aria-label="Decrement value"
                class="_button_e43be3"
              >
                 - 
              </button>
              <span
                aria-label="Counter value"
                class="_value_e43be3"
              >
                0
              </span>
              <button
                aria-label="Increment value"
                class="_button_e43be3"
              >
                 + 
              </button>
            </div>
            <div
              class="_row_e43be3"
            >
              <input
                aria-label="Set increment amount"
                class="_textbox_e43be3"
                value="2"
              />
              <button
                aria-label="Add increment amount"
                class="_button_e43be3"
              >
                Add
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
